Cinema.Web/Controllers/TicketController.cs

```diff
@@ -1,147 +1,151 @@
+    {
+        private readonly BusinessLogic.Interfaces.ITicketService _ticketService;
+        private readonly IMapper _mapper;
+        private readonly IHttpClientFactory _clientFactory;
+        private readonly ICinemaSearcherClient _cinemaSearcherClient;
+
+        public TicketController(BusinessLogic.Interfaces.ITicketService ticketService, IMapper mapper, IHttpClientFactory clientFactory, ICinemaSearcherClient cinemaSearcherClient)
+        {
+            _ticketService = ticketService;
+            _mapper = mapper;
+            _clientFactory = clientFactory;
+            _cinemaSearcherClient = cinemaSearcherClient;
+        }
+
+        /// <summary>
+        /// Get specific ticket.
+        /// </summary>
+        /// <param name="ticketId">Ticket Id.</param>
+        /// <returns>Place by id.</returns>
+        /// <response code="404">If ticket is not found.</response>
+        [HttpGet]
+        [Route("{ticketId}")]
+        [ProducesResponseType(typeof(IActionResult), 200)]
+        [ProducesResponseType(404)]
+        public async Task<IActionResult> Get(Guid ticketId)
+        {
+            var ticket = await _ticketService.GetAsync(ticketId);
+
+            return Ok(ticket);
+        }
+
+        /// <summary>
+        /// Get all tickets.
+        /// </summary>
+        /// <returns>All tickets.</returns>
+        /// <response code="404">If tickets are not found.</response>
+        [HttpGet]
+        [ProducesResponseType(typeof(IActionResult), 200)]
+        [ProducesResponseType(404)]
+        public async Task<IActionResult> GetAll()
+        {
+            var tickets = await _ticketService.GetAllAsync();
+
+            return Ok((tickets));
+        }
+
+        /// <summary>
+        /// Get all tickets.
+        /// </summary>
+        /// <returns>All tickets.</returns>
+        /// <response code="404">If tickets are not found.</response>
+        [HttpGet]
+        [Route("/price-list")]
+        [ProducesResponseType(typeof(IActionResult), 200)]
+        [ProducesResponseType(404)]
+        public async Task<IActionResult> GetPriceList()
+        {
+            var tickets = await _ticketService.GetAllAsync();
+            var filteredTickets = _mapper.Map<IEnumerable<Ticket>, List<FilteredTicket>>(tickets);
+
+            return Ok(filteredTickets);
+        }
+
+        /// <summary>
+        /// Get all tickets by search query.
+        /// </summary>
+        /// <returns>All tickets by search query.</returns>
+        /// <response code="404">If tickets are not found.</response>
+        [HttpGet]
+        [Route("/api/ticket/search")]
+        [ProducesResponseType(typeof(IActionResult), 200)]
+        [ProducesResponseType(404)]
+        public async Task<IActionResult> GetAllBySearchQuery([FromQuery] TicketSearchModel ticketSearchModel)
+        {
+            IEnumerable<Ticket> tickets = await _cinemaSearcherClient.GetBySearchQuery(Request.QueryString);
```

Comment:
If I understand correctly here we get data from service and send it into client
But as I know we need select data from db also
Is it correct?